test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNetCore/AspNetCoreMvc2Tests.cs

```diff
@@ -1,166 +1,24 @@
-#if NETCOREAPP2_1
-
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.IO;
-using System.Linq;
-using System.Net;
-using System.Threading;
-using Datadog.Trace.TestHelpers;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Datadog.Trace.ClrProfiler.IntegrationTests.AspNetCore
 {
-    public class AspNetCoreMvc2Tests : TestHelper
+    public class AspNetCoreMvc2Tests : AspNetCoreMvcTestBase
     {
-        private static readonly string _topLevelOperationName = "aspnet-coremvc.request";
-
-        private static readonly List<WebServerSpanExpectation> _expectations = new List<WebServerSpanExpectation>()
-        {
-            CreateTopLevelExpectation(url: "/", httpMethod: "GET", httpStatus: "200", resourceUrl: "/"),
-            CreateTopLevelExpectation(url: "/delay/0", httpMethod: "GET", httpStatus: "200", resourceUrl: "delay/{seconds}"),
-            CreateTopLevelExpectation(url: "/api/delay/0", httpMethod: "GET", httpStatus: "200", resourceUrl: "api/delay/{seconds}"),
-            CreateTopLevelExpectation(url: "/status-code/203", httpMethod: "GET", httpStatus: "203", resourceUrl: "status-code/{statusCode}"),
-            // TODO: The below test succeeds in IISExpress, but fails in self host when expecting a status code of 500.
-            CreateTopLevelExpectation(
-                url: "/bad-request",
-                httpMethod: "GET",
-                httpStatus: null,
-                resourceUrl: "bad-request",
-                additionalCheck: span =>
-                {
-                    var failures = new List<string>();
-                    if (span.Tags[Tags.ErrorMsg] != "This was a bad request.")
-                    {
-                        failures.Add($"Expected specific exception within {span.Resource}");
-                    }
-
-                    return failures;
-                }),
-        };
-
         public AspNetCoreMvc2Tests(ITestOutputHelper output)
             : base("AspNetCoreMvc2", output)
         {
         }
 
+#if NETCOREAPP2_1
```

Comment:
Note that versions of ASP.NET Core MVC before 3.0 can be run on .NET Framework (does not require .NET Core).

Reply:
Okay. I maintained what was there for now if that's okay. I hope to follow up with more stuff.